docs/src/main/asciidoc/tls-registry-reference.adoc

```diff
@@ -883,3 +883,160 @@ On Mac, you can use the following command:
+== Automatic certificate management with Let's Encrypt
+
+https://letsencrypt.org[Let's Encrypt] is a free, automated certificate authority provided by https://www.abetterinternet.org/[Internet Security Research Group].
+
+Let's Encrypt uses https://datatracker.ietf.org/doc/html/rfc8555[Automated certificate management environment (ACME) protocol] to support an automatic certificate issuance and renewal. Please read https://letsencrypt.org/docs/[Let's Encrypt documentation] to learn more about Let's Encrypt and ACME.
+
+TLS registry project provides a CLI ACME client to issue and renew Let's Encrypt certificates.
+Your application uses TLS registry to resolve ACME protocol challenges.
+
+Follow the steps below to have your Quarkus application prepared and automatically updated with new and renewed Let's Encrypt certificates.
+
+[[lets-encrypt-prerequisites]]
+=== Prerequisites
+
+Make sure that a fully resolvable DNS domain name is available and can be used to access your application.
+This domain name is used for creating a Let's Encrypt account, and supporting Let's Encrypt ACME challenges to prove that you own this domain.
+You can use https://ngrok.com/[Ngrok] to start experimenting with the Quarkus Let's Encrypt ACME feature, see <<lets-encrypt-ngrok>> section below for more information.
+
+Your Quarkus HTTPS application must use a _build-time_ property to enable a Let's Encrypt ACME challenge route:
+
+[source, properties]
+----
+quarkus.tls.lets-encrypt.enabled=true
+----
+
+The TLS registry can manage the challenge process from either the main HTTP interface or from the management interface.
+Using a management interface is **strongly** recommended to let Quarkus deal with ACME challenge configuration separately to the main application's deployment and security requirements:
+
+[source, properties]
+----
+quarkus.tls.lets-encrypt.enabled=true
+quarkus.management.enabled=true
+----
+
+The challenge itself is served from the primary HTTP interface (the one accessible from your DNS domain name).
+
+IMPORTANT: Do not start your application yet.
+
+=== Use CLI to prepare your application
+
+Before you request a Let's Encrypt certificate, you must run TLS registry Let's Encrypt CLI `prepare` command to prepare your application:
+
+[source, shell]
+----
+quarkus tls lets-encrypt prepare --domain=<domain-dns-name>
+----
+
+[IMPORTANT]
+----
+Make sure you run a prepare command in the root directory of your application.
+----
+
+The `prepare` command does the following:
+
+- Creates a `.letsencrypt` folder in your application's root directory
+- Creates a self-signed domain certificate and private key for your application configured in the previous <<lets-encrypt-prerequisites>> step to be able to start and accept HTTPS requests.
+- Create a `.env` configuration file in your application's root directory configure the application to use the self-signed domain certificate and private key (until we get the Let's Encrypt certificate).
```

Comment:
:+1:

Reply:
Though it is not quite clear how this `.env` file aligns with properties below referring to the cert and key files. may be worth expanding a bit on it